src/recon_buildblock/niftypet_projector/ProjectorByBinNiftyPETHelper.cxx

```diff
@@ -0,0 +1,1157 @@
+            for (int x = min_indices[3]; x <= max_indices[3]; x++) {
+                // Convert the stir 3d index to a NiftyPET 1d index
+                np_z = unsigned(z - min_indices[1]);
+                np_y = unsigned(y - min_indices[2]);
+                np_x = unsigned(x - min_indices[3]);
+                np_1d = convert_niftypet_im_3d_to_1d_idx(np_x,np_y,np_z);
+                np_vec[np_1d] = stir[z][y][x];
+            }
+        }
+    }
+}
+
+void
+ProjectorByBinNiftyPETHelper::
+convert_image_niftyPET_to_stir(DiscretisedDensity<3,float> &stir, const std::vector<float> &np_vec)
+{
+    // Get the dimensions of the input image
+    Coordinate3D<int> min_indices;
+    Coordinate3D<int> max_indices;
+    int stir_dim[3];
+    get_stir_indices_and_dims(stir_dim,min_indices,max_indices,stir);
+
+    // NiftyPET requires the image to be (z,x,y)=(SZ_IMZ,SZ_IMX,SZ_IMY)
+    // which at the time of writing was (127,320,320).
+    const int np_dim[3] = {SZ_IMZ,SZ_IMX,SZ_IMY};
+    check_im_sizes(stir_dim,np_dim);
+    check_voxel_spacing(stir);
+
+    // Copy data from NiftyPET to STIR image
+    unsigned np_z, np_y, np_x, np_1d;
+    for (int z = min_indices[1]; z <= max_indices[1]; z++) {
+        for (int y = min_indices[2]; y <= max_indices[2]; y++) {
+            for (int x = min_indices[3]; x <= max_indices[3]; x++) {
+                // Convert the stir 3d index to a NiftyPET 1d index
+                np_z = unsigned(z - min_indices[1]);
+                np_y = unsigned(y - min_indices[2]);
+                np_x = unsigned(x - min_indices[3]);
+                np_1d = convert_niftypet_im_3d_to_1d_idx(np_x,np_y,np_z);
+                stir[z][y][x] = np_vec[np_1d];
+            }
+        }
+    }
+
+    // After the back projection, we enforce a truncation outside of the FOV.
+    // This is because the NiftyPET FOV is smaller than the STIR FOV and this
+    // could cause some voxel values to spiral out of control.
+    // truncate_rim(stir,17);
+}
+
+void
+get_vals_for_proj_data_conversion(std::vector<int> &sizes, std::vector<int> &segment_sequence,
+                                  int &num_sinograms, int &min_view, int &max_view,
+                                  int &min_tang_pos, int &max_tang_pos,
+                                  const ProjDataInfo& proj_data_info, const std::vector<float> &np_vec)
+{
+    const ProjDataInfoCylindricalNoArcCorr * info_sptr =
+            dynamic_cast<const ProjDataInfoCylindricalNoArcCorr *>(&proj_data_info);
+    if (is_null_ptr(info_sptr))
+        error("ProjectorByBinNiftyPETHelper: only works with cylindrical projection data without arc-correction");
+
+    const int max_ring_diff   = info_sptr->get_max_ring_difference(info_sptr->get_max_segment_num());
+    const int max_segment_num = info_sptr->get_max_segment_num();
+
+    segment_sequence.resize(unsigned(2*max_ring_diff+1));
+    sizes.resize(unsigned(2*max_ring_diff+1));
```

Comment:
```suggestion
    segment_sequence.resize(unsigned(2*max_segment_num+1));
    sizes.resize(unsigned(2*max_segment_num+1));
```

Reply:
I'm creating a function for the `segment_sequence` in #506, but it will do the same thing of course